anycloud/cli/src/deploy/mod.rs

```diff
@@ -1313,16 +1313,19 @@ where
     // it's ok to leave out the newline chars, since `sp.println` will insert
     // those for us
     let new_lines = logs.split("\n").skip(lines.len()).collect::<Vec<_>>();
+    // print latest line if any
+    if lines.len() > 0 {
+      if let Some(last_line) = lines.get(lines.len() - 1) {
+        sp.println(last_line);
+      }
+    }
     // update the spinner and lines above the spinner
     new_lines
       .into_iter()
-      .filter(|line| !line.is_empty())
-      .for_each(|line| {
-        if let Some(last_line) = lines.get(lines.len() - 1) {
-          sp.println(last_line);
-        }
-        sp.set_message(line);
-        lines.push(line.to_string());
+      .filter(|new_line| !new_line.is_empty())
+      .for_each(|new_line| {
+        sp.set_message(new_line);
+        lines.push(new_line.to_string());
```

Comment:
this changes the behavior and does not do what is expected.

Say `lines` is empty and `new_lines` contains this:
```
"hello"
"hey"
"hi"
```

the result after the `for_each` is done will be:
```
<spinner> hi
```
when it should be
```
hello
hey
<spinner> hi
```

Reply:
True, with this change the behaviour is the following that is not expected:
![image](https://user-images.githubusercontent.com/5898639/121712545-24908900-cadc-11eb-86f5-9c37bad06806.png)

Restored it

Reply:
With the initial fix works as expected:
![image](https://user-images.githubusercontent.com/5898639/121713018-a8e30c00-cadc-11eb-8a64-4c2d8e1ba1bd.png)